.github/workflows/azure-e2e-test.yml

```diff
@@ -0,0 +1,151 @@
+name: azure-e2e-test
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  PODVM_GALLERY_NAME: "podvm_gallery3"
+  PODVM_IMAGE_DEF_NAME: "podvm_image3"
+  PODVM_IMAGE_VERSION: "0.0.3"
+  PODVM_IMAGE_NAME: "peerpod-image-0.0.1"
```

Comment:
nit: resource names look a bit arbitrary

Reply:
Planning to shift this to github secrets as well.

Reply:
I wouldn't move this to secrets, those are not secret values. it will make the test harder to read and understand, the strings will be masked in the output and changes between runs are obscured.